### DISCUSSION TITLE
metrics: micrometer vs mp metrics

### DISCUSSION THREAD

**Discussion by guykatz** [quarkusio/quarkus#24703] [Q&A]
Hi guys;
the quarkus guides state:
_[Micrometer](https://quarkus.io/guides/micrometer) is the recommended approach to metrics for Quarkus. Use the SmallRye Metrics extension when it is required to retain MicroProfile specification compatibility._
can someone ellaborate on why this is the reccomended approach? seems like quarkus tries to stay MP favored in all other areas so I am curious. 
also, are there future plans in MP metrics spec that will tilt the decision back to using MP metrics? (i assume something is missing)
thanks!

[a comment by an automated account is not shown]

**Reply by ebullient** [reply]
This was explained here: 
https://quarkus.io/blog/micrometer-metrics/

and here:
https://groups.google.com/g/microprofile/c/k85_2po0jh4/m/Iyh-VcYVAQAJ

Outside of compatibility where required, we will not be encouraging use of MP Metrics.

**Reply by guykatz** [reply]
> This was explained here: https://quarkus.io/blog/micrometer-metrics/
> 
> and here: https://groups.google.com/g/microprofile/c/k85_2po0jh4/m/Iyh-VcYVAQAJ
> 
> Outside of compatibility where required, we will not be encouraging use of MP Metrics.

thanks!
exactly what I was looking for
